src/Forms/TOTPForm.php

```diff
@@ -2,44 +2,49 @@
 
 namespace ElliotSawyer\TOTPAuthenticator;
 
-use Firesphere\BootstrapMFA\Forms\BootstrapMFALoginForm;
 use SilverStripe\Control\RequestHandler;
 use SilverStripe\Forms\FieldList;
 use SilverStripe\Forms\FormAction;
 use SilverStripe\Forms\HiddenField;
 use SilverStripe\Forms\PasswordField;
+use SilverStripe\Forms\RequiredFields;
+use SilverStripe\Security\LoginForm;
 
 /**
  * Class TOTPForm
  * @package ElliotSawyer\TOTPAuthenticator
  */
-class TOTPForm extends BootstrapMFALoginForm
+class TOTPForm extends LoginForm
 {
     /**
      * TOTPForm constructor.
      * @param RequestHandler|null $controller
-     * @param null $validator
      * @param string $name
+     * @param null|TOTPAuthenticator $authenticator
      */
     public function __construct(
         RequestHandler $controller = null,
-        $validator = null,
-        $name = self::DEFAULT_NAME
+        $name = self::DEFAULT_NAME,
+        $authenticator = null
     ) {
         $this->controller = $controller;
         $fields = $this->getFormFields();
         $actions = $this->getFormActions();
+        $validator = RequiredFields::create(['token']);
 
-        parent::__construct($controller, $validator, $name, $fields, $actions);
+        parent::__construct($controller, $name, $fields, $actions, $validator);
+        $this->setAuthenticatorClass(get_class($authenticator));
     }
 
     /**
      * @return FieldList|static
      */
     public function getFormFields()
     {
-        $fields = FieldList::create();
-        $fields->push(PasswordField::create('token', _t(self::class . '.TOTPCODE', 'TOTP Code')));
+        $fields = FieldList::create([
+            PasswordField::create('token', _t(self::class . '.TOTPCODE', 'TOTP Code')),
+            HiddenField::create('AuthenticationMethod', $this->authenticator_class)
```

Comment:
FYI I've made a PR to add a `getAuthenticatorClass()` method to use here instead of accessing this property directly: https://github.com/silverstripe/silverstripe-framework/pull/8571

Won't be useful until this module requires 4.4 or newer though